tests/unit_tests/test_wave_06.py

```diff
@@ -257,15 +259,103 @@ def test_swap_best_by_category_no_other_match_is_false():
     # Act
     result = tai.swap_best_by_category(
         other_vendor=jesse,
-        my_priority="Electronics",
-        their_priority="Decor"
+        my_priority="Electronics",   # tai priority
+        their_priority="Decor"       # jesse priority
     )
 
-    raise Exception("Complete this test according to comments below.")
-    # *********************************************************************
-    # ****** Complete Assert Portion of this test **********
-    # *********************************************************************
+    assert not result
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert tai.inventory == [item_c, item_b, item_a]
+    assert jesse.inventory == [item_f, item_e, item_d]
+
     # Assertions should check:
     # - That result is falsy
     # - That tai and jesse's inventories are the correct length
     # - That all the correct items are in tai and jesse's inventories
+
+def test_item_has_valid_age():
+    # Arrange
+    age = (2024,7,8)
+    # Act
+    item = Item(age=age)
+    # Assert
+    assert item.age
+    assert item.age == datetime(*age).date()
+
+def test_swap_by_newest_returns_true():
+    # Arrange
+    item_a = Decor(age=(2024,9,6)) # newest
+    item_b = Electronics(age=(2023,6,6))
+    item_c = Decor(age=(2022,12,6))   
+    tai = Vendor(
+        inventory=[item_c, item_b, item_a]
+    )
+
+    item_d = Clothing(age=(2024,9,1))
+    item_e = Decor(age=(2023,9,6))
+    item_f = Clothing(age=(2024,9,8))   # newest
+    jesse = Vendor(
+        inventory=[item_f, item_e, item_d]
+    )
+    # Act
+    exchanged = tai.swap_by_newest(jesse)
+    
+    # Assert
+    assert exchanged
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert tai.inventory == [item_c, item_b, item_f]
+    assert jesse.inventory == [item_e, item_d,item_a]
+    
+def test_swap_by_newest_swaps_first_newest_in_inventory():
+    # Arrange
+    item_a = Decor(age=(2024,9,6)) # newest
+    item_b = Electronics(age=(2023,6,6))
+    item_c = Decor(age=(2024,9,6)) # newest
+    tai = Vendor(
+        inventory=[item_c, item_b, item_a]
+    )
+
+    item_d = Clothing(age=(2024,9,1))
+    item_e = Decor(age=(2023,9,6))
+    item_f = Clothing(age=(2024,9,8))   # newest
+    jesse = Vendor(
+        inventory=[item_f, item_e, item_d]
+    )
+    # Act
+    print(f"{tai.inventory=}")
+    print(f"{jesse.inventory=}")
+    exchanged = tai.swap_by_newest(jesse)
+    print(f"{tai.inventory=}")
+    print(f"{jesse.inventory=}")
+    
+    # Assert
+    assert exchanged
+    assert len(tai.inventory) == 3
+    assert len(jesse.inventory) == 3
+    assert tai.inventory == [ item_b, item_a, item_f]
+    assert jesse.inventory == [ item_e, item_d, item_c ]
```

Comment:
Nit: inconsistent whitespaces around square brackets. 
```suggestion
    assert tai.inventory == [item_b, item_a, item_f]
    assert jesse.inventory == [item_e, item_d, item_c]
```